manifests/init.pp

```diff
@@ -20,6 +21,7 @@
         default: { fail("unsupported osfamily: ${::osfamily}") }
       }
     }
+    $require_package = Package[$real_package_name]
```

Comment:
Puppet variables are immutable. `$require_package` was already set on line 13, and catalog compilation will fail if `$manage_package` is evaluated to true.

You can change the code to something like this to avoid this issue:

```puppet
if $manage_package {
  # ...
  $require_package = Package[$real_package_name]
} else {
  $require_package = undef
}
```

But maybe selectors can help you write more concise and less complex code, something like this (all files managed by the module are managed after all packages managed by the module):

```puppet
Package <| tags == 'patterndb' |> -> File <| tags == 'patterndb' |>
```